web/src/main/proto/spine/web/subscriptions.proto

```diff
@@ -0,0 +1,181 @@
+
+import "spine/options.proto";
+
+option (type_url_prefix) = "type.spine.io";
+option java_package = "io.spine.web";
+option java_outer_classname = "SubscriptionsProto";
+option java_multiple_files = true;
+
+import "google/protobuf/any.proto";
+import "google/protobuf/duration.proto";
+import "google/protobuf/timestamp.proto";
+import "spine/base/error.proto";
+import "spine/client/subscription.proto";
+import "spine/core/ack.proto";
+
+// A subscription initiated by a web client.
+//
+// The subscription may have `extra` properties required by the web client.
+//
+message WebSubscription {
+
+    client.Subscription subscription = 1;
+
+    repeated google.protobuf.Any extra = 2;
+}
+
+// A request to subscribe to the given topics.
+message Subscribe {
+
+    // Topics to which to subscribe.
+    repeated client.Topic topic = 1;
+
+    // The requested minimum lifetime of the subscriptions.
+    //
+    // As the clients cannot be trusted to reliably cancel all the subscriptions, the server puts
+    // a limit on the subscription lifetime. When a subscription exceeds the expected lifetime,
+    // the server *may* cancel it.
+    //
+    // In order to prolong subscription lifetime, the client must send the `KeepUp` requests.
+    //
+    // This value is a requested value. The server may set another lifetime to the subscriptions.
+    // The actual "valid through" time is returned by the server in response to this request.
+    //
+    google.protobuf.Duration lifespan = 2;
+}
+
+// The response to the `Subscribe` request.
+message Subscribing {
+
+    // The subscription results list.
+    //
+    // For each subscription topic, this is either a `Subscription`, or an error which prohibits
+    // creating such a subscription.
+    //
+    // The order of the results corresponds to the order of topics in the `Subscribe` request.
+    //
+    repeated SubscriptionOrError result = 1;
+
+    // For all the subscriptions in the response, this is the time after which the subscription
+    // will expire.
+    //
+    // An expired subscription may be cancelled by the server.
+    //
+    google.protobuf.Timestamp valid_thru = 2;
+}
+
+// Either a `spine.client.Subscription` or an error that occurred when creating the subscription.
+message SubscriptionOrError {
+
+    oneof payload {
+        WebSubscription subscription = 1;
+        base.Error error = 2;
+    }
+}
+
+// A request to prolong the life of the given subscriptions.
+message KeepUp {
+
+    // IDs of the subscriptions to keep up.
+    repeated client.SubscriptionId subscription = 1;
+
+    // How much time does the client require.
+    //
+    // The client requests that the server add this amount of time to the subscription life.
+    // The server may add this time or another duration. The updated subscription expiration date
+    // is provided in the response.
+    //
+    // If the client wishes to *reduce* the lifetime of the subscription, it can do so by providing
```

Comment:
Oh my god :) What if we submit something really-really negative? I.e. `Duration.of(-65535, YEAR)`?

I would not support negative values ever, as it will bring much more issues down the road. And — I suspect — the increased volume of the code to handle the edge cases. It is going to out-weight the benefits 
— from all cases, it's probably <1% that will use these negative values.

Also, negative `Duration`s looks really weird.